### PR TITLE
feat: Trigger playbooks on events

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: events/playbook.go [hunk 1]
```diff
@@ -1,11 +1,157 @@
...
+		}
+
+		if playbook.Spec.Approval == nil || playbook.Spec.Approval.Approvers.Empty() {
+			run.Status = models.PlaybookRunStatusScheduled
+		}
+
+		switch specEvent.Class {
+		case "canary":
+			// What should be the component or config id?
+			if ok, err := matchResource(eventResource.Check.Labels, map[string]any{"check": eventResource.Check}, playbook.Spec.On.Canary); err != nil {
```
Comment (moshloop) [inline events/playbook.go:69]: why not just take an eventResource object e.g. matchResource(eventResource, playbook.Spec.On.Canary)

File: fixtures/playbooks/scale-deployment.yaml [hunk 2]
```diff
@@ -3,6 +3,18 @@ kind: Playbook
 metadata:
   name: scale-deployment
 spec:
+  'on':
```
Comment (moshloop) [inline fixtures/playbooks/scale-deployment.yaml:6]: Add new fixtures for the event handling

File: events/playbook.go (hunk 1 above)
Comment (adityathebe) [reply events/playbook.go:69]: `matchResource` does not know whether it should use the labels from the check or the component.

File: events/playbook.go [hunk 3]
```diff
@@ -104,36 +107,38 @@ func logToJobHistory(ctx *api.Context, playbookID, err string) {
 // for the given labels and cel env.
 func matchResource(labels map[string]string, celEnv map[string]any, matchFilters []v1.PlaybookEventDetail) (bool, error) {
 	for _, mf := range matchFilters {
+		var (
```
Comment (moshloop) [inline events/playbook.go:110]: Can we not just use a `continue outer` with an `outer: ` label ?

**Pull request merged**